### DISCUSSION TITLE
How to have a list of stings as a function schema argument?

### DISCUSSION THREAD

**Discussion by Tillaert** [metosin/malli#807] [Q&A] [answered]
I am trying to validate the input of a function, where the input is a vector of strings:

`(f ["foo" "bar" "baz"])`

whoever:

`{:malli/schema [:=> [:cat [+: :string]]]}` will check varargs, and not a vector of strings.

How do you check for a list of strings?

**Comment by ikitommi** [comment]
Try `[:cat [:vector :string]]`

**Comment by ikitommi** [comment, marked as answer]
... `[:+ :string]` is a "sequence of strings", but nested sequence schemas are inlined. You can escape the inlining also with using a non-sequence schema, `:schema` being the simplest way to do that:

```clj
;; int and 1+ strings
[:cat :int [:+ :string]]

;; int and sequence of 1+ strings
[:cat :int [:schema [:+ :string]]]
```
